### PR TITLE
feat: reclaim superseded and deleted document blobs from ContentStore

### LINKED ISSUE #687: Reclaim superseded and deleted document blobs from ContentStore
Parent tracker: #656 (item 1).

## Problem

`ContentStore` paths are keyed by document version. A new version no longer overwrites the previous blob, and deleting database records does not remove their stored document content. Superseded versions and content belonging to deleted documents can therefore accumulate indefinitely.

This is separate from query-key garbage collection in #669: this issue owns files in `ContentStore`, not reachability of SQLite query rows.

## Desired outcome

- Reclaim blobs for superseded document versions.
- Reclaim blobs for documents no longer retained in the database.
- Preserve blobs still referenced by an active document/version.
- Make cleanup safe across the app and Share Extension sharing the app-group container.
- Add automated coverage for retention and reclamation.

The implementation may use reference counting or a reachability sweep; this issue deliberately does not prescribe the mechanism.

### PR DESCRIPTION
**Pull request by paulgessinger**
### Reclaim superseded and deleted document blobs

`ContentStore` paths are keyed by document version, so a new version leaves the old file on disk and deleting a document leaves its download orphaned. Nothing ever removed either.

A reachability sweep, not reference counting: the database already records each cached document's current version, so one read (`retainedContentVersions()`, all servers at once) is the whole reachable set. Everything else under the store is deleted: superseded versions, documents the reconcile pruned, and whole servers whose connection was removed.

Safe against the app group being shared with the Share Extension:
- only the canonical names the store writes itself are ever deletion candidates, so partial or foreign files are untouched;
- anything modified within a one-hour grace window is skipped, covering the gap between a blob landing and the row that references it;
- empty directories go via `rmdir(2)`, which fails rather than deleting a subtree a concurrent writer just filled, and vanishing files are tolerated.

Runs from the per-server reconcile after the delete sweep, throttled to once an hour (the reconcile itself runs every 300 s and the input rarely changes); the steady-state cost is one directory listing per server. Also exposed unthrottled in the debug menu, which reports files and bytes reclaimed.

Retention is current-version-only because nothing in the app downloads any other version today; if version browsing lands, `retainedContentVersions()` must widen.

**Follow-up on record.** There is no inventory of blobs: the store is filesystem-only, the sidecar holds just `modified` and the write time, and no size is recorded anywhere. That is fine for an hourly sweep but not for a proactive blob fill or for on-disk usage figures (#673), which want "which documents lack a blob" and "how many bytes" as cheap queries. The intended follow-up is a `content_blob` table (server, document, version, kind, bytes, written at) maintained by whichever process writes the blob, with the filesystem staying the source of truth and this sweep becoming the reconciler that repairs the table against disk. It comes with blob fill or #673, whichever lands first.

Tests cover retention, reclamation of superseded versions, deleted documents and removed servers, the grace period, and a file vanishing mid-sweep.

Closes #687. Part of #656 (item 1).

### Manual test checklist

- [ ] Open a couple of documents so their PDFs download, then Settings → Debug → "Reclaim unused document files": expect 0 removed and N kept as too recent.
- [ ] On the server, add a new version to a document you already opened. Pull to refresh that document, come back after an hour (or force-quit and return later), run the debug reclaim: the old version's blob is reported removed.
- [ ] Delete a document on the server that you had downloaded. Pull to refresh the list, wait past the grace period, run the debug reclaim: its file is removed.
- [ ] Remove a server connection that had downloads; after the grace period the debug reclaim removes that server's whole blob directory.
- [ ] In Airplane Mode, open a document you kept: it still renders from cache.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_018agkypG5HAxHiYjnrbDmp9

### CODE REVIEW

**Comment by paulgessinger** [comment]
@codex

File: AppShared/Sources/AppShared/Networking/ServerSession.swift [hunk 1]
```diff
@@ -472,6 +484,27 @@ public final class ServerSession {
...
+    // network. After the delete sweep on purpose — a document pruned above
+    // loses its downloaded file in the same pass rather than a pass later.
+    //
+    // Deliberately not a `sweep`: reclamation is bookkeeping, not freshness, so
+    // it neither advances the "something refreshed" count nor marks the pass
+    // failed. Nothing the user sees depends on it having run.
+    if !result.cancelled, shouldReclaimContent() {
+      lastContentReclaim = Date()
+      do {
+        let report = try await backend.reclaimDocumentContent()
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply AppShared/Sources/AppShared/Networking/ServerSession.swift]: 🤖 _Posted by Claude on Paul's behalf. Analysis by Claude; the duplicated-walk path was traced by reading, not measured._

Valid — fixed in `c1b120a8`.

Traced it: `lastContentReclaim` was a per-instance `@ObservationIgnored private var`, and `SyncEngine.runSweep` reaches the reclaim branch once per session via `runAction` → `sync(stored:phases:)` → `runSync` → `reconcileDocuments(force: true)` → `runReconcile`. On a cold launch every session starts with a nil stamp, so S configured servers each performed the full `retainedContentVersions()` read plus a whole-directory walk of the app-group blob store — and passes 2..S could only find nothing. It also contradicted the contract the PR writes two files over in `CachingRepository`: "Callers therefore need not (and must not) run this once per server."

The stamp is now `static`. `ServerSession` is `@MainActor`, so that is concurrency-safe without a lock, and because the stamp is taken *before* the `await`, a second session arriving in the same launch flurry sees it and skips — so no separate in-flight guard is needed either.

File: AppShared/Sources/AppShared/Document/DocumentStore.swift [hunk 2]
```diff
@@ -896,6 +896,17 @@ extension DocumentStore {
...
   }
+
+  /// Debug / maintenance: drop downloaded document files that no cached document
+  /// version references any more, keeping everything still reachable.
+  ///
+  /// The same sweep the reconcile runs, minus its hourly throttle — the point of
+  /// the debug affordance is to see the effect now, on demand.
+  @discardableResult
+  public func reclaimDocumentContent() async throws -> ContentStore.ReclaimReport {
+    guard let backend = session?.backend else { return ContentStore.ReclaimReport() }
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply AppShared/Sources/AppShared/Document/DocumentStore.swift:907]: 🤖 _Posted by Claude on Paul's behalf. Analysis by Claude; the narrowing below was established by reading `ServerSession.invalidate()` and `retainedContentVersions()`, not by running the app._

Real, but narrower than stated — and deferred by decision rather than fixed here. Tracked as item 39 in #656.

The mechanism is as you describe: `invalidate()` sets `built = nil` when a connection row disappears, so `DocumentStore.reclaimDocumentContent()`'s `session?.backend` guard fails and it returns an empty `ReclaimReport()` — the debug alert then says "Removed 0 files", indistinguishable from "nothing to reclaim".

Where the consequence is overstated: `retainedContentVersions()` computes the reachable set across **every** server in one read, and the blob store is one shared app-group directory, so a removed server's blobs are collected by the next *surviving* server's reconcile. The genuine exposure is the last connection being removed — then nothing sweeps until one is added back — plus the debug affordance no-opping whenever the store is not activated.

Not refactoring it now: the opportunistic PDF/blob download work will rework this GC anyway, and wants a path that runs over all elements rather than per-server ones. Extracting a process-wide owner today would be churn against code about to be replaced, and it overlaps the `runReconcile` region PR #709 also edits. The sibling finding (the duplicated per-server walk) *was* worth fixing standalone and is done in `c1b120a8`.

#656 item 39 records the concrete shape for when that work lands: lift the throttle, single-flight and sweep into an owner that outlives any one session — a `ContentReclaimer` in `Persistence` reached through `ServerSessionRegistry`, which also gets it a `swift test` target, since `AppShared` has none.
